packages/ui-react/src/components/Form/FormSection.tsx

```diff
@@ -147,9 +147,9 @@ export function FormSection<TData extends GenericFormValues>({
   );
 
   return (
-    <div className={className}>
+    <section aria-labelledby={sectionId} className={className}>
```

Comment:
I noticed that the id that `useOpaqueId` generates, could contain spaces, which is not valid in an `id` attribute.
I suggest making a change in `generateId` which is used by 'useOpaqueId' to remove spaces.

And maybe convert `.` (dots) to `_`? I do not know the root cause of this, but i noticed this in a snapshot-file.

Since we're only using lowercase characters within our id I also suggest applying `toLowerCase` too ;)

![Screenshot 2024-01-10 at 10 34 27](https://github.com/Videodock/ott-web-app/assets/1019129/fcf5fa8a-3d1a-4348-96fd-757897deee32)

Reply:
The 'root cause' is this: `const sectionId = useOpaqueId(label);` (FormSection L45)
We should do something like this:
`const sectionId = useOpaqueId(label.replace(/\s/g, '').toLowerCase());`

Or do that to both the prefix and suffix in `useOpaqueId`, although I think that's a bit too much 'fallbacking'.

Reply:
If we also want to remove the dots (as Mike pointed out), it could be: `replace(/[\s.]+/g, '').toLowerCase()`

Reply:
I do not know the root cause of the dots, but I do of the spaces. ;)

Why not fix it in the function itself as I believe there could be future more instances where we do not want spaces (instead of specifically and repeatedly remove them in every `useOpaqueId` call)

Reply:
Yes okay, let's fix it there. I like your replace at the end, after the join. But I think a regex is needed also to perform it globally (replaceAll is not fully supported).

```javascript
 return [prefix, Math.round(randomId), suffix].filter(Boolean).join('_').replace(/[\s.]+/g, '').toLowerCase();
```